plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/launcher/PipelineArgumentsTabTest.java

```diff
@@ -68,11 +76,22 @@ public void testGetName() {
 
     // https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/2165
     @Test
-    public void testInitializeForm_noExceptionForNonAccessibleProject() throws CoreException {
+    public void testInitializeForm_noExceptionForNonAccessibleProject()
+        throws CoreException, InvocationTargetException, InterruptedException {
       IWorkspaceRoot workspaceRoot = mock(IWorkspaceRoot.class);
       when(workspaceRoot.getProject(anyString())).thenReturn(mock(IProject.class));
 
       ILaunchConfigurationDialog dialog = mock(ILaunchConfigurationDialog.class);
+      doAnswer(new Answer<Void>() {
+        @Override
+        public Void answer(InvocationOnMock invocation)
+            throws InvocationTargetException, InterruptedException {
+          IRunnableWithProgress runnable = invocation.getArgumentAt(2, IRunnableWithProgress.class);
+          runnable.run(new NullProgressMonitor());
```

Comment:
Interesting… my only concern here is that any SWT work performed by this runnable will succeed in tests.

Reply:
Would this be worth extracting into a helper class?